build.rs

```diff
@@ -70,45 +81,66 @@ fn setup_mbed_crypto(mbed_config: &MbedConfig) {
     let metadata = package
         .metadata
         .expect("Cargo.toml does not contain package metadata.");
-    let plasma_config = get_value_from_table(&metadata, CONFIG_TABLE_NAME);
-    let mbed_version = get_configuration_string(&plasma_config, MBED_CRYPTO_VERSION_KEY);
+    let parsec_config = get_value_from_table(&metadata, CONFIG_TABLE_NAME);
+    let mbed_version = get_configuration_string(&parsec_config, MBED_CRYPTO_VERSION_KEY);
 
     let mut run_script = ::std::process::Command::new(SETUP_MBED_SCRIPT_PATH);
     run_script.arg(mbed_version).arg(mbed_path);
 
-    if let Some(mbed_compiler) = &mbed_config.mbed_compiler {
-        run_script.arg(format!("CC={}", mbed_compiler));
+    let toolchain;
+    let mbed_compiler;
+    let mbed_archiver;
+    if std::env::var("TARGET").unwrap() == "aarch64-unknown-linux-gnu" {
+        toolchain = mbed_config.aarch64_unknown_linux_gnu.as_ref().unwrap();
+        mbed_compiler = toolchain
+            .mbed_compiler
+            .clone()
```

Comment:
Do you need to clone here? Since you only use the value in an if-else, you should be able to just extract it out of `toolchain`

Reply:
If I remove the `clone` I have the following error:
```
error[E0507]: cannot move out of `toolchain.mbed_compiler` which is behind a shared reference
  --> build.rs:95:25
   |
95 |           mbed_compiler = toolchain
   |  _________________________^
96 | |             .mbed_compiler
   | |__________________________^ move occurs because `toolchain.mbed_compiler` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait
help: consider borrowing the `Option`'s content
   |
95 |         mbed_compiler = toolchain
96 |             .mbed_compiler.as_ref()
   |
```
If I use `as_ref` however then in the `unwrap_or` I have on one side a `&String` and one the other a `&str`. I did not find a good way to convert both to the same type so I just assumed using `clone` was acceptable here because of that 😃